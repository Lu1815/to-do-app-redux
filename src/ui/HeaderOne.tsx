import clsx from 'clsx'

interface IHeaderOneProps {
    content?: string,
    color?: string,
    weigth?: string
}

const HeaderOne = ({ content, color, weigth }: IHeaderOneProps) => {
  return (
    <p className={
        clsx(
            'text-lg',
            (color ?? 'text-black'),
            (weigth ?? 'font-normal')
        )
    }>
        { content ?? 'Normal Text' }
    </p>
  )
}

export default HeaderOne