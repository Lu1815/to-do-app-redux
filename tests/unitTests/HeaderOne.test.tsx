import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import HeaderOne from '../../src/ui/HeaderOne';

describe('HeaderOne Component', () => {
  test('renders HeaderOne with default props', () => {
    render(<HeaderOne />);
    expect(screen.getByText('Normal Text')).toBeInTheDocument();
    expect(screen.getByText('Normal Text')).toHaveClass('text-lg');
    expect(screen.getByText('Normal Text')).toHaveClass('text-black');
    expect(screen.getByText('Normal Text')).toHaveClass('font-normal');
  });

  test('renders HeaderOne with specified content', () => {
    render(<HeaderOne content="Custom Content" />);
    expect(screen.getByText('Custom Content')).toBeInTheDocument();
  });

  test('renders HeaderOne with specified color and weight', () => {
    render(<HeaderOne color="text-red-500" weigth="font-bold" />);
    const header = screen.getByText('Normal Text');
    expect(header).toHaveClass('text-lg');
    expect(header).toHaveClass('text-red-500');
    expect(header).toHaveClass('font-bold');
  });

  test('matches snapshot', () => {
    const { container } = render(<HeaderOne content="Snapshot Test" color="text-blue-500" weigth="font-semibold" />);
    expect(container).toMatchSnapshot();
  });
});
