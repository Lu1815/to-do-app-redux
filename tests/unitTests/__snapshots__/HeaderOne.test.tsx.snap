// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderOne Component matches snapshot 1`] = `
<div>
  <p
    class="text-lg text-blue-500 font-semibold"
  >
    Snapshot Test
  </p>
</div>
`;
